.github/workflows/parachain.yml

```
@@ -16,116 +16,37 @@ on:
-      - name: install protoc
-        uses: arduino/setup-protoc@v1
-        with:
-          repo-token: ${{ secrets.GITHUB_TOKEN }}
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: nightly-2023-04-22
-          target: wasm32-unknown-unknown
-      - name: cargo check
-        uses: actions-rs/cargo@v1
-        with:
-          command: check
-          toolchain: nightly-2023-04-22
-          args: >-
-            --manifest-path parachain/Cargo.toml
-            --workspace
-            --features runtime-benchmarks
-  test:
-    needs: check
-    runs-on: snowbridge-runner
-    env:
-      CARGO_INCREMENTAL: 0
-      RUST_BACKTRACE: 1
-      RUSTFLAGS: -C debuginfo=1
-      SKIP_WASM_BUILD: 1
-    steps:
-      - uses: actions/checkout@v2
-        with:
-          ref: ${{ github.head_ref }}
-      - uses: actions/cache@v1
-        with:
-          path: |
-            ~/.cargo/registry
-            ~/.cargo/git
-          key: ${{ runner.os }}-cargo-${{ hashFiles('parachain/Cargo.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-cargo-
-      - uses: arduino/setup-protoc@v1
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: nightly-2023-04-22
-          target: wasm32-unknown-unknown
-      # Increase stack limit for beacon light client tests
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/install_toolchain
       - run: sudo prlimit --pid $$ --stack=32768
-      # Run tests for everything except the beacon light client
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: >-
-            --manifest-path parachain/Cargo.toml
-            --workspace
-            --features runtime-benchmarks
-            --exclude snowbridge-query-events
-            --exclude snowbridge-ethereum-beacon-client
-          toolchain: nightly-2023-04-22
-      # Run tests for beacon light client, excluding benchmark tests
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: >-
-            --manifest-path parachain/Cargo.toml
-            --package snowbridge-ethereum-beacon-client
-          toolchain: nightly-2023-04-22
-      # Run benchmark tests for beacon light client
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
-          args: >-
-            --manifest-path parachain/Cargo.toml
-            --release
-            --features runtime-benchmarks
-            --package snowbridge-ethereum-beacon-client
-          toolchain: nightly-2023-04-22
+      - name: cargo check
+        run: cargo check --manifest-path parachain/Cargo.toml --workspace --features runtime-benchmarks
+      - name: cargo test
+        run: cargo test --manifest-path parachain/Cargo.toml --workspace --features runtime-benchmarks --exclude snowbridge-query-events --exclude snowbridge-ethereum-beacon-client
```

This is easier to read:
```suggestion
      - name: Tests for everything except the beacon light client
        run: >| 
          cargo test 
          --manifest-path parachain/Cargo.toml
          --workspace
          --features runtime-benchmarks
          --exclude snowbridge-query-events
          --exclude snowbridge-ethereum-beacon-client
```